packages/core/src/domain/Operator.ts

```diff
@@ -6,10 +6,13 @@ import type Type from './Type';
  * Represents an operator that can be
  * applied to values in some group methods.
  */
-export default interface Operator<TArgs extends UnknownArgs, TValue> {
+export default interface Operator<
+  TArgs extends UnknownArgs = UnknownArgs,
+  TValue = unknown,
+> {
   readonly type: OperatorType;
   readonly allowedTypes: Type[];
   apply<TApplyArgs extends TArgs, TApplyValue extends TValue>(
-    ...args: TApplyArgs
-  ): TApplyValue;
+    args: TApplyArgs,
+  ): TApplyValue[];
```

Comment:
question(blocking): why does `apply` return array? in which cases?

Reply:
Correct, fixed